core-ui/src/components/Predefined/Create/Certificates/Certificates.create.js

```diff
@@ -1,63 +1,189 @@
-        content: t('certificates.create.messages.success'),
-      });
-      LuigiClient.linkManager()
-        .fromContext('namespace')
-        .navigate(`/certificates/details/${certificate.name}`);
-    } catch (e) {
-      console.error(e);
-      notification.notifyError({
-        title: t('certificates.create.messages.failure'),
-        content: e.message,
-      });
-      return false;
-    }
-  };
+  const [withCSR, setWithCSR] = useState(false);
+  const [existingSecret, setExistingSecret] = useState(false);
+  const [csrIsEncoded, setCsrIsEncoded] = useState(false);
 
   return (
-    <CreateForm
+    <ResourceForm
+      pluralKind="certificates"
+      singularName={t('certificates.name_singular')}
       title={t('certificates.create.title')}
-      simpleForm={
-        <SimpleForm certificate={certificate} setCertificate={setCertificate} />
-      }
-      advancedForm={
-        <AdvancedForm
-          certificate={certificate}
-          setCertificate={setCertificate}
-        />
-      }
       resource={certificate}
       setResource={setCertificate}
-      toYaml={toYaml}
-      fromYaml={fromYaml}
-      onCreate={createCertificate}
       onChange={onChange}
       formElementRef={formElementRef}
-    />
+      createUrl={`/apis/cert.gardener.cloud/v1alpha1/namespaces/${namespace}/certificates/`}
+    >
+      <ResourceForm.K8sNameField
+        propertyPath="$.metadata.name"
+        kind={t('certificates.name_singular')}
+        setValue={name => {
+          jp.value(certificate, '$.metadata.name', name);
+          jp.value(
+            certificate,
+            "$.metadata.labels['app.kubernetes.io/name']",
+            name,
+          );
+          setCertificate({ ...certificate });
+        }}
+      />
+      <ResourceForm.KeyValueField
+        advanced
+        propertyPath="$.metadata.labels"
+        title={t('common.headers.labels')}
+        className="fd-margin-top--sm"
+      />
+      <ResourceForm.KeyValueField
+        advanced
+        propertyPath="$.metadata.annotations"
+        title={t('common.headers.annotations')}
+      />
+      <ResourceForm.FormField
+        label={t('certificates.with-csr')}
+        tooltipContent={t('certificates.tooltips.with-csr')}
+        input={() => (
+          <div className="csr-header">
+            <Switch
+              compact
+              onChange={e => setWithCSR(!withCSR)}
+              checked={withCSR}
+            />
+            {withCSR && (
+              <Button
+                compact
+                className="csr-encode-toggle"
+                option="transparent"
+                glyph={csrIsEncoded ? 'show' : 'hide'}
+                onClick={() => setCsrIsEncoded(!csrIsEncoded)}
+              >
+                {csrIsEncoded
+                  ? t('secrets.buttons.decode')
+                  : t('secrets.buttons.encode')}
+              </Button>
+            )}
+          </div>
+        )}
+      />
+      {withCSR && (
```

Comment:
The CSR doesn't work properly - If I type something into the `CSR` input, then change the switch, the previously typed-in value is still in the yaml